### PR TITLE
The security dog - Muhtar. The third try.

### PR DESCRIPTION
**Pull request by NikaVichy**
## About the PR

The third try. Literally skill issue.

- Added Muhtar and sprites for him
- Added Muhtar's spawner to every station except for CentComm and Reach
- Resprited tanks and masks to fit on Muhtar

## Why / Balance

- Imported from ss13.
- You'll need 10h of sec. dep. to take a ghostrole.
- Speed higher then human one: 6 tiles for sprint, 4.5 for walking.
- Has 30 stamina damage - not much to easilly drop a nukie from dark, just enough to stop a criminal running away. Tho stil waiting for your opinion.
- Has 5 slash and 5 piercing damage. Was playtested, not much, not op at all. Same as previous.
- Can't slip on anything. Never seen a dog that slips on puddles or smthng like this.
- He can drag objects or just move things around with his mouth.
- 75 damage till crit, 140 till death
- 50 damage before half slowing down. That's a trained dog, it has to deal with damage.
- 150 stamina damage till stamcrit. Well, it is tougher than any being, cuz it is still trained.

## Technical details

- none

## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add:  Added Muhtar - the security dog! Woof!

### CODE REVIEW

**Comment by NikaVichy** [comment]
finally.

**Comment by Simyon264** [comment]
FYI: There is a freeze on new pets currently. See #8524.
> New pets. While fluff is nice we have enough cargo-obtainable fluff animals and department mascots, to add new animals they must have something interesting like new AI behaviors or spawn in proc gen maps.

**Comment by NikaVichy** [comment]
> FYI: There is a freeze on new pets currently. See #8524.
> 
> > New pets. While fluff is nice we have enough cargo-obtainable fluff animals and department mascots, to add new animals they must have something interesting like new AI behaviors or spawn in proc gen maps.

Tho animals are freezed, it's a new role for players, moreover, more RP. If it's gonna be closed - whatever, gonna add it to my own station.

**Comment by NikaVichy** [comment]
I've wanted to replace with it Shiva, but my friend told me, in ss13 there were both Muhtar and Shiva.

**Comment by IlyaElDunaev** [comment]
I've played as a security dog, an interesting experience. This diversifies the gameplay.

**Comment by lzk228** [comment]
We dont need more pets, at least new features for pets will be added.

**Review by VasilisThePikachu — CHANGES_REQUESTED** [review]
1. Like a lot of people said. Pets are fully frozen. You are welcome to keep this PR open. However, it will most likely not be merged any time soon. (#8524)
2. I recommend dropping the mapping changes and instead using migrations.yml found in Resources to replace Shiva. (a mapping maintainer may correct me here. This may be fine)
3. Fix tests

**Comment by NikaVichy** [comment]
> 1. Like a lot of people said. Pets are fully frozen. You are welcome to keep this PR open. However, it will most likely not be merged any time soon. ([Current Freezes #8524](https://github.com/space-wizards/space-station-14/issues/8524))
> 2. I recommend dropping the mapping changes and instead using migrations.yml to replace shiva. Or if you wanna keep shiva, drop the map changes and let mappers do it themselves. (a mapping maintainer may correct me here. This may be fine)
> 3. Fix tests

Yes, thanks, I'll do that.

**Comment by NikaVichy** [comment]
Also noticed that I've basically erased some meta. Gonna take some time to fix this.

**Comment by Errant-4** [comment]
You could also put it in a K9 Krate TM that can be ordered through cargo (do we currently have a way to limit cargo orders by time or presence of a mob on the station?)

**Comment by Vaaankas** [comment]
> We dont need more pets, at least new features for pets will be added.

Just remove both useless sec pets for this actually fun one.

**Comment by NikaVichy** [comment]
> You could also put it in a K9 Krate TM that can be ordered through cargo (do we currently have a way to limit cargo orders by time or presence of a mob on the station?)

I'll fix the tests first, wait for approve and only then i'll do it. Don't really want to waste my time and still get rejected.

**Comment by Errant-4** [comment]
> > You could also put it in a K9 Krate TM that can be ordered through cargo (do we currently have a way to limit cargo orders by time or presence of a mob on the station?)
> 
> I'll fix the tests first, wait for approval and only then i'll do it. Don't really want to waste my time and still get rejected.

I meant, as a way to avoid "no more roundstart ghostroles"

**Comment by NikaVichy** [comment]
> > > You could also put it in a K9 Krate TM that can be ordered through cargo (do we currently have a way to limit cargo orders by time or presence of a mob on the station?)
> > 
> > 
> > I'll fix the tests first, wait for approval and only then i'll do it. Don't really want to waste my time and still get rejected.
> 
> I meant, as a way to avoid "no more roundstart ghostroles"



> > > You could also put it in a K9 Krate TM that can be ordered through cargo (do we currently have a way to limit cargo orders by time or presence of a mob on the station?)
> > 
> > 
> > I'll fix the tests first, wait for approval and only then i'll do it. Don't really want to waste my time and still get rejected.
> 
> I meant, as a way to avoid "no more roundstart ghostroles"

image
Still isn't a way.

**Comment by NikaVichy** [comment]
Fixed tests and meta. Seems good now.
Well, I've decided not to replace Shiva or McGriff, cuz thief still has objectives on them. Removing Shiva or McGriff to add another content isn't a way either. 
Now it's fate in your hands.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml
Comment (IProduceWidgets) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2599]: NanoTrasen

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml
Comment (IProduceWidgets) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2667]: Why false? Disarming seems pretty in-line for a police dog.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml
Comment (IProduceWidgets) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2679]: This seems pretty fast. My intuition says that it should only be slightly faster than the default sprint speed of 4.5 otherwise its just not going to ever be possible to get away from and ruin chases.

The walk speed should probably just be the default 2.5 because you want a fine-input mode.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml
Comment (IProduceWidgets) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2698]: Similar issue as the speeds. Without slips its just never possible to get away from and will mean chases are not possible/fun.

**Comment by IProduceWidgets** [comment]
Just a disclaimer that I don't have merge perms, so those reviews are just my opinions.

**Comment by IProduceWidgets** [comment]
One thing I know which has been weird is four legged sprites can look funny when moving. There is the functionality for movement spritesets which could be neat.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Don't add custom sprites for a singular mob, it's not worth the maintenance burden.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml
Comment (NikaVichy) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2679]: Yeah, gonna make it 5.5 or something so.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml
Comment (NikaVichy) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2667]: > Why false? Disarming seems pretty in-line for a police dog.

Playtesters' told me that it was hard to hit anybody with LMB. Moreover it has 30 stamina damage, it wont be that hard to stamcrit anybody.

Comment (NikaVichy) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2667]: And I actualy dont see a way of any dog disarming anybody without harming them. Is it even possible when human is resisting?

**Comment by NikaVichy** [comment]
> Don't add custom sprites for a singular mob, it's not worth the maintenance burden.

The only reason I did it is incorrect sprite displayment. Both tanks and masks were out of place. Tanks were levitating and masks were on eyes, it looked ridiculous.

**Comment by NikaVichy** [comment]
> One thing I know which has been weird is four legged sprites can look funny when moving. There is the functionality for movement spritesets which could be neat.

None of the four-legged animals have animation now.

**Comment by FairlySadPanda** [comment]
Also just a random contrib

Think a ghost role that aids sec is a bit risky because by necessity it has to be filled by a ghost, who will have access to round knowledge. Sec doc with validhunting rights + ghostrole = high likellihood for abuse

That said this is a great sprite and a police dog makes much more sense than a police spider

**Comment by VasilisThePikachu** [comment]
> Also just a random contrib
> 
> Think a ghost role that aids sec is a bit risky because by necessity it has to be filled by a ghost, who will have access to round knowledge. Sec doc with validhunting rights + ghostrole = high likellihood for abuse
> 
> That said this is a great sprite and a police dog makes much more sense than a police spider

It's not even a valid hunting issue and stuff. Just that I (personally) don't want more round-start ghost roles in general.

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Pull request closed without merging**
